### PR TITLE
openimageio2: 2.2.17.0 -> 2.4.6.1

### PR DESCRIPTION
**Pull request by risicle**
###### Description of changes

This was surprisingly painful.

https://nvd.nist.gov/vuln/detail/CVE-2022-36354
https://nvd.nist.gov/vuln/detail/CVE-2022-38143
https://nvd.nist.gov/vuln/detail/CVE-2022-41639
https://nvd.nist.gov/vuln/detail/CVE-2022-41649
https://nvd.nist.gov/vuln/detail/CVE-2022-41684
https://nvd.nist.gov/vuln/detail/CVE-2022-41794
https://nvd.nist.gov/vuln/detail/CVE-2022-41837
https://nvd.nist.gov/vuln/detail/CVE-2022-41838
https://nvd.nist.gov/vuln/detail/CVE-2022-41977
https://nvd.nist.gov/vuln/detail/CVE-2022-41981
https://nvd.nist.gov/vuln/detail/CVE-2022-41988
https://nvd.nist.gov/vuln/detail/CVE-2022-41999
https://nvd.nist.gov/vuln/detail/CVE-2022-43592
https://nvd.nist.gov/vuln/detail/CVE-2022-43593
https://nvd.nist.gov/vuln/detail/CVE-2022-43594
https://nvd.nist.gov/vuln/detail/CVE-2022-43595
https://nvd.nist.gov/vuln/detail/CVE-2022-43596
https://nvd.nist.gov/vuln/detail/CVE-2022-43597
https://nvd.nist.gov/vuln/detail/CVE-2022-43598
https://nvd.nist.gov/vuln/detail/CVE-2022-43599
https://nvd.nist.gov/vuln/detail/CVE-2022-43600
https://nvd.nist.gov/vuln/detail/CVE-2022-43601
https://nvd.nist.gov/vuln/detail/CVE-2022-43602
https://nvd.nist.gov/vuln/detail/CVE-2022-43603

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by panicgh** [comment]
minizip-ng: #205689. Similar but not the same.

File: pkgs/applications/graphics/openimageio/2.x.nix [hunk 1]
```diff
@@ -16,13 +16,13 @@
 
 stdenv.mkDerivation rec {
   pname = "openimageio";
-  version = "2.2.17.0";
+  version = "2.4.6.0";
```
Comment (panicgh) [inline pkgs/applications/graphics/openimageio/2.x.nix:19]: There is already a more recent tag, but it is not mentioned on the release page: https://github.com/OpenImageIO/oiio/tags

```suggestion
  version = "2.4.6.1";
```

(Don't forget to change the commit message and PR title accordingly.)

File: pkgs/applications/graphics/openimageio/2.x.nix [hunk 2]
```diff
@@ -16,13 +16,13 @@
...
 
   src = fetchFromGitHub {
     owner = "OpenImageIO";
     repo = "oiio";
-    rev = "Release-${version}";
-    sha256 = "0jqpb1zci911wdm928addsljxx8zsh0gzbhv9vbw6man4wi93h6h";
+    rev = "v${version}";
+    sha256 = "sha256-KXndraB7uZ0ZMcLLvZObzyNxWcDPfAPtkFVmWTKS+TQ=";
```
Comment (panicgh) [inline pkgs/applications/graphics/openimageio/2.x.nix:25]: For 2.4.6.1:

```suggestion
    sha256 = "sha256-oBICukkborxXFHXyM2rIn5qSbCWECjwDQI9MUg13IRU=";
```

File: pkgs/development/libraries/opencolorio/default.nix [hunk 3]
```diff
@@ -52,15 +52,19 @@ stdenv.mkDerivation rec {
...
+      openexr_3
+    ];
 
-    cmakeFlags = [
-      "-DOCIO_INSTALL_EXT_PACKAGES=NONE"
-    ] ++ lib.optional (!pythonBindings) "-DOCIO_BUILD_PYTHON=OFF"
-      ++ lib.optional (!buildApps) "-DOCIO_BUILD_APPS=OFF";
+  cmakeFlags = [
+    "-DOCIO_INSTALL_EXT_PACKAGES=NONE"
+  ] ++ lib.optional (!pythonBindings) "-DOCIO_BUILD_PYTHON=OFF"
```
Comment (panicgh) [inline pkgs/development/libraries/opencolorio/default.nix:66]: Most tests should work, except for the GPU tests. (see below)

```suggestion

    # GPU test fails with: freeglut (GPU tests): failed to open display ''
    "-DOCIO_BUILD_GPU_TESTS=OFF"
  ] ++ lib.optional (!pythonBindings) "-DOCIO_BUILD_PYTHON=OFF"
```

File: pkgs/development/libraries/opencolorio/default.nix [hunk 4]
```diff
@@ -52,15 +52,19 @@ stdenv.mkDerivation rec {
...
+    ++ lib.optional (!buildApps) "-DOCIO_BUILD_APPS=OFF";
 
   # TODO Investigate this: Python and GPU tests fail to load libOpenColorIO.so.2.0
   # doCheck = true;
```
Comment (panicgh) [inline pkgs/development/libraries/opencolorio/default.nix:70]:
```suggestion
  doCheck = true;
  # numpy enables many more tests, increases coverage
  checkInputs = lib.optionals pythonBindings [ python3Packages.numpy ];
```

**Review by panicgh — CHANGES_REQUESTED** [review]

**Comment by panicgh** [comment]
This is not part of your original changes, so I cannot create "suggestions".

You may consider to replace the `postPatch` in https://github.com/NixOS/nixpkgs/pull/208013/files#diff-1986409fd6747fb3793cc3a0cb1ea89b703e2e04372f66df8d0ad6daf0706400R72-R77 with an upstream patch:

```patch
--- a/pkgs/development/libraries/opencolorio/default.nix
+++ b/pkgs/development/libraries/opencolorio/default.nix
@@ -43,6 +52,12 @@ stdenv.mkDerivation rec {
       revert = true;
       sha256 = "sha256-0DF+lwi2nfkUFG0wYvL3HYbhZS6SqGtPWoOabrFS1Eo=";
     })
+
+    (fetchpatch {
+      name = "pkg-config-absolute-path.patch";
+      url = "https://github.com/AcademySoftwareFoundation/OpenColorIO/pull/1720.patch";
+      sha256 = "sha256-IhL8p6OQPEeYGOechqbAHTPCKXG7VVHRIoCiHkA85HE=";
+    })
   ];

   nativeBuildInputs = [ cmake ];
```

File: pkgs/applications/graphics/openimageio/2.x.nix (hunk 1 above)
Comment (risicle) [reply pkgs/applications/graphics/openimageio/2.x.nix:19]: Yeah I was a little unclear as to whether they consider that actually "released".

File: pkgs/development/libraries/opencolorio/default.nix (hunk 3 above)
Comment (risicle) [reply pkgs/development/libraries/opencolorio/default.nix:66]: Ok let's see how this goes :crossed_fingers:

**Comment by risicle** [comment]
Hmmmmm opencolorio tests appear to have some precision issues on aarch64.

**Comment by risicle** [comment]
Looks like I might need to apply https://github.com/OpenImageIO/oiio/pull/3722 to fix aarch64 too.

**Comment by risicle** [comment]
@panicgh note the fix I added for `minizip-ng` here, having missed it in your PR.

**Review by panicgh — CHANGES_REQUESTED** [review]
The link redirects to https://sites.google.com/site/openimageio/home but I think it is good to keep the current one in case upstream decides to move elsewhere. But it can be made https and similar to opencolorio.

```diff
--- a/pkgs/applications/graphics/openimageio/2.x.nix
+++ b/pkgs/applications/graphics/openimageio/2.x.nix
@@ -70 +70 @@ stdenv.mkDerivation rec {
-    homepage = "http://www.openimageio.org";
+    homepage = "https://openimageio.org";
```

**Comment by panicgh** [comment]
Basically it LGTM:

Result of `nixpkgs-review pr 208013` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>blender</li>
    <li>blender-hip</li>
    <li>linphone</li>
    <li>minizip-ng</li>
    <li>opencolorio</li>
    <li>openimageio2</li>
    <li>osl</li>
    <li>python310Packages.bpycv</li>
    <li>python39Packages.bpycv</li>
  </ul>
</details>

But opencolorio fails for pkgsCross.aarch64-multiplatform.opencolorio. Maybe this would work on a native aarch64-linux and it is an upstream cross-compilation issue.

```
CMake Error at /nix/store/blqlkgmvs12h8m8rhbv7pmyj7jyxxp37-cmake-3.24.3/share/cmake-3.24/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find Python (missing: Python_EXECUTABLE Python_INCLUDE_DIRS
  Interpreter Development.Module) (found version "3.10.9")
Call Stack (most recent call first):
  /nix/store/blqlkgmvs12h8m8rhbv7pmyj7jyxxp37-cmake-3.24.3/share/cmake-3.24/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  /nix/store/blqlkgmvs12h8m8rhbv7pmyj7jyxxp37-cmake-3.24.3/share/cmake-3.24/Modules/FindPython.cmake:561 (find_package_handle_standard_args)
  share/cmake/modules/FindExtPackages.cmake:91 (find_package)
  CMakeLists.txt:261 (include)
```

**Comment by risicle** [comment]
`pkgsCross.aarch64-multiplatform.opencolorio` doesn't build for me on master either though.

**Comment by risicle** [comment]
Also, this branch _does_ successfully build `pkgsCross.aarch64-multiplatform.opencolorio` on my x86_64 nixos machine.

**Review by rastertail — APPROVED** [review]

**Review by panicgh — APPROVED** [review]
I approve this definitely for x86_64-linux, and for aarch64-linux I trust your tests. However, I cannot test it for aarch64-linux, since master and this PR's branch fail for me for `pkgsCross.aarch64-multiplatform.opencolorio` on x86_64-linux. At the same time, Hydra seems to be fine with aarch64-linux on master. So it seems to be a problem on my side, not sure what that is.

**Comment by risicle** [comment]
@ofborg build pkgsCross.aarch64-multiplatform.opencolorio

I'm confused about this, because it builds for me.

**Comment by risicle** [comment]
Innnnteresting, fails on ofborg too.

**Comment by risicle** [comment]
@ofborg build pkgsCross.aarch64-multiplatform.opencolorio

**Comment by risicle** [comment]
Ok, my naive attempts at fixing this failed and TBH I don't think we've actually had `opencolorio` cross-building before, so in the interests of getting the security fix out I'm going to merge this...

**Comment by risicle** [comment]
For 22.11 I'm considering bumping openimageio, but leaving it `opencolorio_1`-based, via a backport of #208429 if that ever gets reviewed.

**Pull request merged**